setup.sh

```diff
@@ -1,14 +1,23 @@
 #!/usr/bin/env bash
 
-docker compose --ansi never up -d --build 2>& 1> /dev/null
-docker compose cp proxy:/usr/bin/caddy /tmp/caddy
+docker compose up -d 2>/dev/null
 
-XDG_DATA_HOME=/var/lib/docker/volumes/dev_caddy_data/_data
+docker compose exec proxy wget -q -O- --post-data='"0.0.0.0:2019"' --header='Content-Type:application/json' 'http://127.0.0.1:2019/config/admin/listen'
 
-sudo XDG_DATA_HOME=${XDG_DATA_HOME} \
-/tmp/caddy start 2&> /dev/null; \
-/tmp/caddy untrust; \
-/tmp/caddy trust; \
-/tmp/caddy stop 2&> /dev/null; \
+docker compose cp proxy:/usr/bin/caddy /tmp/caddy 2>/dev/null
 
-rm /tmp/caddy
+/tmp/caddy trust
+
+# ArchLinux and derivatives, see https://wiki.archlinux.org/title/Transport_Layer_Security#Add_a_certificate_to_a_trust_store
+if command -v trust &> /dev/null; then
+  sudo trust anchor /usr/local/share/ca-certificates/*.crt
+  sudo update-ca-trust
+fi
+
+rm -f /tmp/caddy
+
+# Copy the certificate to a stable location (to simplify mounting on all Docker platforms)
+sudo docker compose cp proxy:/data/caddy/pki/authorities/local/root.crt /etc/ssl/certs/Caddy.crt 2>/dev/null
```

Comment:
<!-- Nitpicks are trivial preference-based requests. These should be non-blocking by nature.  -->
**nitpick (non-blocking):** ‏You should not write in a system directory without checking there is not already a file called  `Caddy.crt`. You might rename this file with something more obvious like `lephare-local.crt`.